### PR TITLE
Fix "Not all auth domains are evaluated for TransportRequests" SG-951

### PR DESCRIPTION
**Pull request by floragunncom**

### CODE REVIEW

File: src/main/java/com/floragunn/searchguard/auth/BackendRegistry.java [hunk 1]
```diff
@@ -330,45 +330,45 @@ public User call() throws Exception {
...
-            try {              
-                
-                if(authenticatedUser == null) {
-                    if(log.isDebugEnabled()) {
-                        log.debug("Cannot authenticate user (or add roles) with ad {} due to user is null, try next", authDomain.getOrder());
-                    }
-                    continue;
-                }
-                
-                if(AdminDNs.isAdmin(authenticatedUser.getName())) {
```
Comment (jochenkressin) [inline src/main/java/com/floragunn/searchguard/auth/BackendRegistry.java]: Was this removed on purpose?

File: src/main/java/com/floragunn/searchguard/auth/BackendRegistry.java [hunk 2]
```diff
@@ -330,45 +330,45 @@ public User call() throws Exception {
...
-                if(log.isDebugEnabled()) {
-                    log.debug("User '{}' is authenticated", authenticatedUser);
-                }
-                return authenticatedUser;
-            } catch (final ElasticsearchSecurityException e) {
-                if(log.isDebugEnabled()) {
-                    log.debug("Cannot authenticate user (or add roles) with ad {} due to {}, try next", authDomain.getOrder(), e.toString());
-                }
+            if(authenticatedUser == null) {
+                /*if(log.isDebugEnabled()) {
```
Comment (jochenkressin) [inline src/main/java/com/floragunn/searchguard/auth/BackendRegistry.java]: Can we remove the commented code?

File: src/main/java/com/floragunn/searchguard/auth/BackendRegistry.java (hunk 1 above)
Comment (floragunncom) [reply src/main/java/com/floragunn/searchguard/auth/BackendRegistry.java]: its still there, just a few lines below

File: src/main/java/com/floragunn/searchguard/auth/BackendRegistry.java (hunk 2 above)
Comment (floragunncom) [reply src/main/java/com/floragunn/searchguard/auth/BackendRegistry.java]: will check

**Pull request merged**